orchestra/cmds/update.py

```diff
@@ -83,33 +92,54 @@ def handle_update(args):
             - `git pull --rebase`, to pull remote changes and apply your commits on top
             - `git push` your changes to the remotes
         """)
+        failed_git_pull_suggestion = failed_git_pull_template.format(formatted_failed_pulls=formatted_failed_pulls)
         logger.error(failed_git_pull_suggestion)
 
 
 def pull_binary_archive(name, config):
     binary_archive_path = os.path.join(config.binary_archives_dir, name)
-    logger.info(f"Pulling binary archive {name}")
+    # This check is to ensude we are called with the path of an existing binary archive
+    # and don't clean/reset orchestra configuration
+    if not is_git_repo(binary_archive_path):
+        raise OrchestraException(f"{binary_archive_path} is not a git repo, aborting")
+    # clean removes untracked files
+    git_clean(binary_archive_path)
+    # reset restores tracked files to their committed version
+    git_reset_hard(binary_archive_path)
     result = git_pull(binary_archive_path)
     return result
 
 
 def clone_binary_archive(name, url, config):
-    logger.info(f"Trying to clone binary archive from remote {name} ({url})")
     binary_archive_path = os.path.join(config.binary_archives_dir, name)
-    env = dict(os.environ)
+    env = os.environ.copy()
     env["GIT_SSH_COMMAND"] = "ssh -oControlPath=~/.ssh/ssh-mux-%r@%h:%p -oControlMaster=auto -o ControlPersist=10"
     env["GIT_LFS_SKIP_SMUDGE"] = "1"
-    result = subprocess.run(["git", "clone", url, binary_archive_path], env=env)
+    result = run_internal_subprocess(
+        ["git", "clone", url, binary_archive_path],
+        environment=env,
+    )
     if result.returncode:
         logger.info(f"Could not clone binary archive from remote {name}!")
 
 
+def git_clean(directory):
+    return run_internal_subprocess(["git", "-C", directory, "clean", "-d", "--force"])
+
+
+def git_reset_hard(directory, ref="master"):
+    return run_internal_subprocess(["git", "-C", directory, "reset", "--hard", ref])
```

Comment:
Note to self: might be necessary to set `GIT_LFS_SKIP_SMUDGE` here to avoid generating network traffic